File: Front end/FoodBox/src/app/common/order-item.ts
import { CartItem } from "./cart-item";

export class OrderItem {
    imageURL: String;
    unitPrice: number;
    quantity:number;
    productId: number;

    constructor(cartItem: CartItem){
        this.imageURL=cartItem.imageURL;
        this.productId=cartItem.id;
        this.quantity=cartItem.quantity;
        this.unitPrice=cartItem.product_Price;
    }
}

File: Back end/FoodBox/src/main/java/com/ecommerce/foodbox/entity/OrderItem.java
package com.ecommerce.foodbox.entity;

import lombok.Getter;

import lombok.Setter;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Table(name="order_item")
@Getter
@Setter
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id")
    private Long id;

    @Column(name="ImageURL")
    private String imageURL;

    @Column(name="unit_price")
    private BigDecimal unitPrice;

    @Column(name="quantity")
    private int quantity;

    @Column(name="Product_id")
    private Long productId;

    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;

}








